src/libraries/Microsoft.Extensions.Options.ConfigurationExtensions/src/ConfigureFromConfigurationOptions.cs

```diff
@@ -12,20 +14,26 @@ namespace Microsoft.Extensions.Options
     /// Configures an option instance by using <see cref="ConfigurationBinder.Bind(IConfiguration, object)"/> against an <see cref="IConfiguration"/>.
     /// </summary>
     /// <typeparam name="TOptions">The type of options to bind.</typeparam>
-    public class ConfigureFromConfigurationOptions<TOptions> : ConfigureOptions<TOptions>
+    public class ConfigureFromConfigurationOptions<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TOptions> : ConfigureOptions<TOptions>
         where TOptions : class
     {
         /// <summary>
         /// Constructor that takes the <see cref="IConfiguration"/> instance to bind against.
         /// </summary>
         /// <param name="config">The <see cref="IConfiguration"/> instance.</param>
+        //Even though TOptions is annotated, we need to annotate as RUC as we can't guarantee that options that are passed in are a nested type.
```

Comment:
This comment is not correct.

"we can't guarantee that options that are passed in are a nested type."

should say

"we can't guarantee properties on referenced types are preserved."